dialogue-serde/src/test/java/com/palantir/conjure/java/dialogue/serde/JacksonEmptyContainerLoaderTest.java

```diff
@@ -0,0 +1,228 @@
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.palantir.conjure.java.serialization.ObjectMappers;
+import com.palantir.dialogue.TypeMarker;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+import java.util.OptionalLong;
+import java.util.Set;
+import javax.annotation.Generated;
+import org.junit.Test;
+
+public class JacksonEmptyContainerLoaderTest {
+
+    private static final ObjectMapper mapper = ObjectMappers.newClientObjectMapper();
+    private final JacksonEmptyContainerLoader emptyContainerDecoder = new JacksonEmptyContainerLoader(mapper);
+
+    @Test
+    public void prove_jackson_doesnt_work_out_of_the_box() throws IOException {
+        assertThat(mapper.readValue("null", Alias1.class)).isNull(); // we want `Alias1.of(Optional.empty())`
+        assertThat(mapper.readValue("null", Alias3.class)).isNull();
+    }
+
+    @Test
+    public void capture_jackson_optional_handling_behaviour() throws IOException {
+        assertThat(mapper.readValue("null", com.google.common.base.Optional.class))
+                .isEqualTo(com.google.common.base.Optional.absent());
+        assertThat(mapper.readValue("null", OptionalInt.class)).isEqualTo(OptionalInt.empty());
+        assertThat(mapper.readValue("null", OptionalDouble.class)).isEqualTo(OptionalDouble.empty());
+        assertThat(mapper.readValue("null", OptionalLong.class)).isEqualTo(OptionalLong.empty());
+    }
+
+    @Test
+    public void capture_jackson_collection_behavior() throws IOException {
+        assertThat(mapper.readValue("null", List.class)).isNull();
+        assertThat(mapper.readValue("null", Map.class)).isNull();
+        assertThat(mapper.readValue("null", Set.class)).isNull();
+    }
+
+    @Test
+    public void http_204_turns_empty_body_into_alias_of_OptionalEmpty() {
+        assertThat(emptyContainerDecoder.getEmptyInstance(new TypeMarker<Alias1>() {}))
+                .isEqualTo(Alias1.of(Optional.empty()));
+    }
+
+    @Test
+    public void http_204_turns_empty_body_into_alias_of_OptionalInt() {
+        assertThat(emptyContainerDecoder.getEmptyInstance(new TypeMarker<Alias2>() {}))
+                .isEqualTo(Alias2.of(OptionalInt.empty()));
+    }
+
+    @Test
+    public void http_204_turns_empty_body_into_alias_of_OptionalDouble() {
+        assertThat(emptyContainerDecoder.getEmptyInstance(new TypeMarker<Alias3>() {}))
+                .isEqualTo(Alias3.of(OptionalDouble.empty()));
+    }
+
+    @Test
+    public void http_204_can_handle_alias_of_alias_of_optional_string() {
+        assertThat(emptyContainerDecoder.getEmptyInstance(new TypeMarker<AliasAlias1>() {}))
+                .isEqualTo(AliasAlias1.of(Alias1.of(Optional.empty())));
+    }
+
+    @Generated("com.palantir.conjure.java.types.AliasGenerator")
```

Comment:
can we avoid checking in generated code? We already generate bindings in other places